### ISSUE TITLE
[Theory or Implementation] Irregular waves 

### ISSUE THREAD

**Issue by timmaflanagan** [WEC-Sim/WEC-Sim#740]
**The development team requests that before submitting a question on the theory and implementation behind WEC-Sim to first please:**
1. Read through our [WEC-Sim Documentation](http://wec-sim.github.io/WEC-Sim/man/theory.html) with additional description found in our [Advanced Features Section](http://wec-sim.github.io/WEC-Sim/man/advanced_features.html)
1. Review MATLAB source code ([functions](https://github.com/WEC-Sim/WEC-Sim/tree/master/source/functions), [objects](https://github.com/WEC-Sim/WEC-Sim/tree/master/source/objects)) and the Simulink blocks ([libraries](https://github.com/WEC-Sim/WEC-Sim/tree/master/source/lib))
1. Review the curated [WEC-Sim Examples](https://github.com/WEC-Sim/WEC-Sim/tree/master/examples) and the [WEC-Sim applications](https://github.com/WEC-Sim/WEC-Sim_Applications)
1. Review the [WEC-Sim issues page](https://github.com/WEC-Sim/WEC-Sim/issues) which allows you to search past resolved issues or questions that might help answer your own question.

**Note: italicized text below is include as an example and should be updated before submission. If you feel any section is not applicable to your request, please replace with 'N/A' rather than delete a section.**

**Is your question request related to a problem? Please describe.**
_A clear and concise description of what the problem is. Ex. I do not understand why I get 'X' result when running 'Y' simulation [...]_
I have recently been running an OSWEC model with irregular waves. From there I develop a power extracted matrix 
![image](https://user-images.githubusercontent.com/79123752/136826935-2e926034-1f66-484e-b0f0-92b10894cb6f.png)
I run each irregular wave condition for 1230s. I have been told that by running it for that length of time I would eliminate any uncertainties and achieve an average power matrix that is indicative of the behaviour of the WEC. Once this power matrix is achieved i obtain a transmission coefficient matrix (kt=sqrt(1-P_intercepted/p_incoming). where p_incoming=((1025*9.81^2)/(64*pi))*H^2*T^2*18:
![image](https://user-images.githubusercontent.com/79123752/136822612-8e2ba08a-5b86-4af8-9f89-3ae9c363408a.png) 
To ensure that the power matrix & transmission matrix are correct, I have been trying to play with the phase seed to see if that would change the solution in any way, and it does. The power matrix is:
![image](https://user-images.githubusercontent.com/79123752/136828256-55a626ef-4802-48cd-92d1-21dea8b9ce44.png)
and the transmission coefficient matrix much more regular:
![image](https://user-images.githubusercontent.com/79123752/136822675-180f3fc5-43ff-4497-bb0d-c02963fa5ed0.png). 
I have been told that the phase seed should not make a difference, and that all the results should converge to the same solution if I run them for long enough (1230s), however this does not seem to be the case. Even if i run the same case (no changes to the input files), I get similar results, but not the exact same results. Could you help me verify that what I have done is correct?

**Describe the theory or implementation approach that you have a question**
_Please clearly describe the assumptions, equations, or code script that you would like clarification. Also, please provide references to other research papers, open-source material, or other information that may be relevant in the WEC-Sim developers understanding your question._

Incoming wave power in the power matrix is calculated using: 
mcr.avgpower = mean(-1*output.ptos.powerInternalMechanics(100:end,5));
mcr.ploss = mean(output.bodies(1).forceMorrisonAndViscous(100:end,5).*output.bodies(1).velocity(100:end,5) + output.bodies(1)forceMorrisonAndViscous(100:end,3).*output.bodies(1).velocity(100:end,3) + output.bodies(1)forceMorrisonAndViscous(100:end,1).*output.bodies(1).velocity(100:end,1)
P_intercepted=mcr.avgpower(imcr)+mcr.ploss(imcr)
P_incoming=(1025*9.81^2)/(64*pi)*waves.H&2*waves.T*18
rampTime=100s;
endTime=1230s;
solver='ode4'
H=0.75:0.5:0.25m;
T=44.5:1:10.5s;
I have included viscous drag coefficients and the characteristic area. I have also specialized the PTO damping to each incoming wave period, and left the stiffness as 0.  

**Personal project context**
_If this request is related to your project, please describe what you are attempting to do in your project so the WEC-Sim development team knows your end goal._

The end goal of my project is to use the transmission coefficient to determine the wave height in the lee of a WEC. once i know that, I can import it into XBeach directly to couple WEC-Sim and XBeach. 

**Describe the type of conclusion or resolution you are looking for**
_[Ex.] I am still unclear on how an irregular wave train is generated within WEC-Sim and would appreciate being pointed to where in the documentation and code where I can find additional information._
I would like to know if I have run my simulations for long enough, or if there seems to be something wrong with my calculations/WEC-Sim files to provide such different results on each run. 

**Additional context**
_Add any other context or screenshots about the feature request here._

**Comment by dforbush2** [comment]
Hi @timmaflanagan 

You're correct that for sufficiently long wave trains, the various phaseSeeds of irregular waves should generate similar statistics over integer numbers of periods. It's important to note that due to some numerical round-offs these may not be _exactly identical_: while it's a little hard to tell from the shifted colormap I think that your power matrix is believably close (correct me if I'm wrong...I can't read all of the numbers) and the distinction looks more severe due to MATLAB colormap defaults

But the transmission coefficient definitely doesn't look right for the first plot. This parameter is not automatically calculated by WEC-Sim...is it possible that you could share your input and processing files? Otherwise it is difficult to help further.

**Comment by timmaflanagan** [comment]
If you need more documentation I can include the entire file for the WEC-Sim trial that I ran.

**Comment by dforbush2** [comment]
Thanks Tim. I'm looking through this right now and I think I have everything working. I am replicating the problem but I'm working on figuring out what exactly is going on.
